### PR TITLE
on slow_queue full -> drop envelope

### PR DESCRIPTION
**Pull request by solarw**

### CODE REVIEW

File: dht/dhtpeer/dhtpeer.go [hunk 1]
```diff
@@ -1169,9 +1169,17 @@ func (dhtPeer *DHTPeer) RouteEnvelope(envel *aea.Envelope) error {
 			routeCount.Dec()
 
 			if !dhtPeer.isClosing {
-				lerror(err).Str("op", "route").Str("addr", target).
-					Msg("while rooting and looking up address on the DHT. moving it to she slow queue")
-				dhtPeer.slow_queue <- envel
+				if len(dhtPeer.slow_queue) >= SlowQueueSize-1 {
+					lerror(err).Str("op", "route").Str("addr", target).
+						Msg("slow queue is full: drop envelope")
+
```
Comment (8ball030) [inline dht/dhtpeer/dhtpeer.go:1175]: how do the agents rect to their envelope being dropped?

Do they try to resend it?

File: dht/dhtpeer/dhtpeer.go [hunk 2]
```diff
@@ -1169,9 +1169,17 @@ func (dhtPeer *DHTPeer) RouteEnvelope(envel *aea.Envelope) error {
...
+				} else {
+					lerror(err).Str("op", "route").Str("addr", target).
+						Msg("while rooting and looking up address on the DHT. moving it to she slow queue.")
+					dhtPeer.slow_queue <- envel
```
Comment (8ball030) [inline dht/dhtpeer/dhtpeer.go:1179]: What happens in the slow queue?
Do we eventually give up? 
Is there a time out?

Can the agent effectively reset their connection when they restart?

File: dht/dhtpeer/dhtpeer.go (hunk 1 above)
Comment (solarw) [reply dht/dhtpeer/dhtpeer.go:1175]: @8ball030 
nope, there are a lot of design issues need to resolve later

File: dht/dhtpeer/dhtpeer.go (hunk 2 above)
Comment (solarw) [reply dht/dhtpeer/dhtpeer.go:1179]: @8ball030 
nope, there are a lot of design issues need to resolve later

**Review by angrybayblade — DISMISSED** [review]

**Review by 8ball030 — APPROVED** [review]

**Review by angrybayblade — APPROVED** [review]

**Pull request merged**